bazel/BUILD.boost

```diff
@@ -1,80 +0,0 @@
-config_setting(
```

Comment:
We don't need this, since we replace it in the previous merged PR: https://github.com/istio-ecosystem/authservice/pull/208. We use github.com/nelhage/rules_boost to manage boost dependencies now.

Reply:
I think this is a leftover :).